packages/hub/src/lib/commit.ts

```diff
@@ -89,18 +132,20 @@ async function* commitIter(params: CommitParams): AsyncGenerator<unknown, Commit
 
 	const lfsShas = new Map<string, string | null>();
 
+	const blobs = await FileBlobs.createBlobMap(params.operations.filter(isFileOperation));
```

Comment:
```suggestion
	// Convert `operation.content` to `Blob`
	const operations = await Promise.all(params.operations.map(op => op.operation === "addOrUpdate" ? {...op, content: await ...} : op));
	
	...
```

No need to create/override a separate `Map` class

Reply:
I did it to avoid casting systematically the `content` to `Blob` but I realize that's a very poor design choice :roll_eyes:

Reply:
(and changing `isFileOperation` too)

Reply:
I tried this but unfortunately since `operation.content` is a `ContentSource`, ts complain that it could be `URL` and not `Blob`.

```js
const operations = await Promise.all(
  params.operations.map(async (operation) => {
    if (isFileOperation(operation) && operation.content instanceof URL) {
      if (operation.content.protocol !== "file:") {
        throw TypeError('Only "file://" protocol is supported for now');
      }

      // Ignore the "file://" at the beginning and the trailing slash
      const lazyBlob = await LazyBlob.create(operation.content.href.slice(7, operation.content.href.length - 1));

      return {
        ...operation,
        content: lazyBlob,
      };
    }

    return operation;
  })
);
```